src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,245 @@
+    {
+        return Iterables.stream(objects).map(AtlasObject::getIdentifier).map(String::valueOf)
+                .collectToList();
+    }
+
+    /**
+     * Checks if the given segment overlaps any of the segments in the given set
+     *
+     * @param segments
+     *            set of segments
+     * @param segment
+     *            any given segment
+     * @return true if the given segment overlaps any of the segments in the set
+     */
+    private boolean isNotSnappedToSegments(final List<Segment> segments, final Segment segment)
+    {
+        return segments.stream().noneMatch(areaSegment -> areaSegment.overlapsShapeOf(segment));
+    }
+
+    /**
+     * Checks if the given edge and area have the same elevation or not.
+     *
+     * @param edge
+     *            any given edge
+     * @param area
+     *            any given area
+     * @return true if the edge and area have the same Layer and Level tag values
+     */
+    private boolean isOfSameElevation(final Edge edge, final Area area)
+    {
+        return LevelTag.getTaggedOrImpliedValue(area, ZERO)
+                .equals(LevelTag.getTaggedOrImpliedValue(edge, ZERO))
+                && LayerTag.getTaggedOrImpliedValue(area, ZERO_LONG)
+                        .equals(LayerTag.getTaggedOrImpliedValue(edge, ZERO_LONG));
+    }
+
+    /**
+     * Checks if given {@link AtlasObject} is a pedestrian area or not. A pedestrian area is an
+     * {@link AtlasObject} with "area=yes" and "highway=pedestrian" tags.
+     *
+     * @param object
+     *            any {@link AtlasObject}
+     * @return true if the object is a pedestrian area
+     */
+    private boolean isPedestrianArea(final AtlasObject object)
+    {
+        return Validators.isOfType(object, AreaTag.class, AreaTag.YES)
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.PEDESTRIAN);
+    }
+
+    /**
+     * Checks if an intersecting edge has a different osm id than the area, does not have pedestrian
+     * highway tags, is car navigable and is of the same level or layer as the area.
+     *
+     * @param edge
+     *            any intersecting edge
+     * @param area
+     *            Area
+     * @return true if the edge is a valid intersecting edge
+     */
+    private boolean isValidIntersectingEdge(final Edge edge, final Area area)
+    {
+        return
+        // We do not want any pedestrian edges that are also pedestrian areas.
+        // Currently, pedestrian ways are ingested as both edges and as areas.
+        edge.getOsmIdentifier() != area.getOsmIdentifier() && edge.isMasterEdge()
+                && !HighwayTag.isPedestrianNavigableHighway(edge)
+                // Valid edge should be a car navigable highway
+                && (HighwayTag.isCarNavigableHighway(edge)
+                        // or a cycleway with no foot access
+                        || (Validators.isOfType(edge, HighwayTag.class, HighwayTag.CYCLEWAY)
```

Comment:
Can you see this list of acceptable intersecting tags changing in the future? If so, would this make sense as a configurable ?

Reply:
For this check, valid edges are non pedestrian navigable ways and I believe we have covered all non pedestrian navigable ways with these validations. If at all there could be changes, it would be in the car navigable highway validation.

Reply:
Okay, if we dont see
`highway=cycleway|service=alley|parking_aisle|driveway|emergency_access|drive_through` ever changing, then I'm okay with it as is